src/quacc/recipes/espresso/phonons.py

```diff
@@ -84,12 +84,201 @@ def phonon_job(
+    want to create your own flow to run both jobs in sequence.
+
+    Parameters
+    ----------
+    prev_dir
+        Outdir of the previously ran q2r.x calculation. This is used to copy
+        the the force constant file.
+    flfrc
+        The name of the file containing the force constants produced by q2r.x.
+        Defaults to "q2r.fc". If you changed it in the q2r.x calculation, you should
+        change it here too.
+    parallel_info
+        Dictionary containing information about the parallelization of the
+        calculation. See the ASE documentation for more information.
+    **calc_kwargs
+        Additional keyword arguments to pass to the Espresso calculator. Set a value to
+        `quacc.Remove` to remove a pre-existing key entirely. See the docstring of
+        [quacc.calculators.espresso.espresso.Espresso][] for more information.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][].
+        See the type-hint for the data structure.
+    """
+
+    calc_defaults = {"input_data": {"input": {"flfrc": flfrc}}}
+
+    prev_dir = {prev_dir: flfrc}
+
+    return base_fn(
+        template=EspressoTemplate("matdyn"),
+        calc_defaults=calc_defaults,
+        calc_swaps=calc_kwargs,
+        parallel_info=parallel_info,
+        additional_fields={"name": "matdyn Phonon"},
+        copy_files=prev_dir,
+    )
+
+
+@flow
+def phonon_dos_flow(
+    atoms: Atoms,
+    job_params: dict[str, Any] | None = None,
+    job_decorators: dict[str, Callable | None] | None = None,
+) -> RunSchema:
+    """
+    Function to carry out a phonon DOS calculation. The phonon calculation is carried out on a coarse q-grid, the force constants are calculated
+    and extrapolated to a finer q-grid, and the phonon DOS is calculated.
+
+    Consists of following jobs that can be modified:
+
+    1. pw.x relaxation
+        - name: "relax_job"
+        - job: [quacc.recipes.espresso.core.relax_job][]
+    2. ph.x calculation
+        - name: "ph_job"
+        - job: [quacc.recipes.espresso.phonons.phonon_job][]
+    3. q2r.x calculation
+        - name: "q2r_job"
+        - job: [quacc.recipes.espresso.phonons.q2r_job][]
+    4. matdyn.x calculation
+        - name: "matdyn_job"
+        - job: [quacc.recipes.espresso.phonons.matdyn_job][]
+
+    Parameters
+    ----------
+    atoms
+        Atoms object to calculate the phonon DOS.
+    job_params
+        Custom parameters to pass to each Job in the Flow. This is a dictionary where the keys are the names of the jobs and the values are dictionaries of parameters.
+    job_decorators
+        Custom decorators to apply to each Job in the Flow. This is a dictionary where the keys are the names of the jobs and the values are decorators.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][].
+        See the type-hint for the data structure.
+    """
```

Comment:
For `@flow`s, I find it a bit nicer to return a dictionary containing _all_ of the results. So that would be something like

```python
{"pw": pw_job_results, "ph": ph_job_results, "fc": fc_job_results, "dos": dos_job_results}
```

with a type hint to clarify this to the user.